testkit/steppedtask.go

```diff
@@ -0,0 +1,379 @@
+		r.t.Fatalf("task '%s' already exists", name)
+	}
+
+	task := newSteppedTask(name, r)
+	r.tasks[name] = task
+	task.setup(fn)
+	return task
+}
+
+// CreateSteppedTestKit creates a new SteppedTestKit
+func (r *SteppedTasksRunner) CreateSteppedTestKit(name string, store kv.Storage) *SteppedTestKit {
+	return newSteppedTestKit(r.t, name, store, r)
+}
+
+func (r *SteppedTasksRunner) recordPath(taskName string, stop *taskstop.StopPoint) {
+	task, ok := r.tasks[taskName]
+	if !ok {
+		r.t.Fatalf("task '%s' not exist", taskName)
+	}
+
+	r.path = append(r.path, struct {
+		task *SteppedTask
+		stop *taskstop.StopPoint
+	}{task: task, stop: stop})
+}
+
+// SteppedCommandTask is a stepped task for sql command
+type SteppedCommandTask struct {
+	t          *testing.T
+	task       *SteppedTask
+	done       bool
+	resultChan chan any
+	result     any
+}
+
+func (t *SteppedCommandTask) updateState() *SteppedCommandTask {
+	switch t.task.currentStop.Name() {
+	case waitingCommandStopPointName:
+		t.done = true
+		select {
+		case result := <-t.resultChan:
+			t.result = result
+		case <-time.After(time.Second * 10):
+			require.FailNow(t.t, "timeout")
+		}
+	case taskstop.DonePointName:
+		t.done = true
+	}
+	return t
+}
+
+// CurrentStop returns the task's current stop
+func (t *SteppedCommandTask) CurrentStop() *taskstop.StopPoint {
+	if t.done {
+		return taskstop.NewStopPoint(taskstop.DonePointName)
+	}
+	return t.task.CurrentStop()
+}
+
+// Continue resumes the task
+func (t *SteppedCommandTask) Continue() *SteppedCommandTask {
+	t.ExpectNotDone()
+	t.task.Continue()
+	return t.updateState()
+}
+
+// IsDone returns whether the current stop is done
+func (t *SteppedCommandTask) IsDone() bool {
+	return t.CurrentStop().Name() == taskstop.DonePointName
+}
+
+// ExpectStoppedAt will check the task stops at the specified stop poin
+func (t *SteppedCommandTask) ExpectStoppedAt(name string) *SteppedCommandTask {
+	require.Equal(t.t, name, t.CurrentStop().Name())
+	return t
+}
+
+// ExpectDone will check the task is done
+func (t *SteppedCommandTask) ExpectDone() *SteppedCommandTask {
+	require.Equal(t.t, taskstop.DonePointName, t.CurrentStop().Name())
+	return t
+}
+
+// ExpectNotDone will check the task not done yet
+func (t *SteppedCommandTask) ExpectNotDone() *SteppedCommandTask {
+	require.NotEqual(t.t, taskstop.DonePointName, t.CurrentStop().Name())
+	return t
+}
+
+// GetResult returns the result of command
+func (t *SteppedCommandTask) GetResult() any {
+	return t.ExpectDone().result
+}
+
+// GetQueryResult returns the result of query
+func (t *SteppedCommandTask) GetQueryResult() *Result {
+	return t.GetResult().(*Result)
+}
+
+const waitingCommandStopPointName = "steppedTestKitWaitingCommand"
+
+// SteppedTestKitCommand is what we want to run for the stepped task
+type SteppedTestKitCommand func(t *testing.T, tk *TestKit, ch *taskstop.Chan) any
+
+// SteppedTestKit is the testkit that can be paused
+type SteppedTestKit struct {
+	t          *testing.T
```

Comment:
Can it embedded `Testkit`, thus you don't need to implement `MustExec` manually?